full-service/src/db/assigned_subaddress.rs

```diff
@@ -249,6 +257,29 @@ impl AssignedSubaddressModel for AssignedSubaddress {
         Ok(assigned_subaddress)
     }
 
+    fn get_for_account_by_index(
+        account_id_hex: &str,
+        index: i64,
+        conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
+    ) -> Result<AssignedSubaddress, WalletDbError> {
+        use crate::db::schema::assigned_subaddresses::dsl::{
+            account_id_hex as dsl_account_id_hex, assigned_subaddresses, subaddress_index,
+        };
+
+        let assigned_subaddress: AssignedSubaddress = match assigned_subaddresses
```

Comment:
The other way to do this is to derive the `public_address_b58` from the account_key - then you can call `AssignedAddress::get`, which looks up by the indexed field.